cmd/image-builder-db-test/main_test.go

```diff
@@ -0,0 +1,142 @@
+// +build integration
+
+package main
+
+import (
+	"context"
+	"fmt"
+        "testing"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v4"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/image-builder/internal/config"
+	"github.com/osbuild/image-builder/internal/db"
+	"github.com/osbuild/image-builder/internal/logger"
+)
+
+const (
+	ANR1 = "000001"
+	ANR2 = "000002"
+	ANR3 = "000003"
+
+	ORGID1= "100000"
+)
+
+func conf(t *testing.T) *config.ImageBuilderConfig {
+	c := config.ImageBuilderConfig{
+		ListenAddress: "unused",
+		LogLevel:      "INFO",
+		MigrationsDir: "/usr/share/image-builder/migrations",
+		PGHost:        "localhost",
+		PGPort:        "5432",
+		PGDatabase:    "imagebuilder",
+		PGUser:        "postgres",
+		PGPassword:    "foobar",
+		PGSSLMode:     "disable",
+	}
+
+	err := config.LoadConfigFromEnv(&c)
+	require.NoError(t, err)
+
+	return &c
+}
+
+func connStr(t *testing.T) string {
+	c := conf(t)
+	return fmt.Sprintf("postgres://%s:%s@%s:%s/%s?sslmode=%s", c.PGUser, c.PGPassword, c.PGHost, c.PGPort, c.PGDatabase, c.PGSSLMode)
+}
+
+func migrateOneStep(t *testing.T) {
+	c := conf(t)
+
+	log, err := logger.NewLogger(c.LogLevel, c.CwAccessKeyID, c.CwSecretAccessKey, c.CwRegion, c.LogGroup)
+	require.NoError(t, err)
+
+	err = db.Migrate(connStr(t), c.MigrationsDir, 1, log)
+	require.NoError(t, err)
+}
+
+func connect(t *testing.T) *pgx.Conn {
+	conn, err := pgx.Connect(context.Background(), connStr(t))
+	require.NoError(t, err)
+	return conn
+}
+
+// after each migration insert some data which follows the schema of that migration
+func insertData1(t *testing.T) {
+	conn := connect(t)
+	defer conn.Close(context.Background())
+
+	insert := "INSERT INTO composes(job_id, request, created_at, account_id, org_id) VALUES ($1, $2, CURRENT_TIMESTAMP, $3, $4)"
+
+	_, err := conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR1, ORGID1)
+	require.NoError(t, err)
+
+	_, err = conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR1, ORGID1)
+	require.NoError(t, err)
+
+	_, err = conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR2, ORGID1)
+	require.NoError(t, err)
+}
+
+func insertData2(t *testing.T) {
+	conn := connect(t)
+	defer conn.Close(context.Background())
+
+	insert := "INSERT INTO composes(job_id, request, created_at, account_number, org_id) VALUES ($1, $2, CURRENT_TIMESTAMP, $3, $4)"
```

Comment:
Why not using `func (db *dB) InsertCompose(jobId, accountId, orgId string, request json.RawMessage) error {` here ?

Reply:
For insertData2 could use that here, as that is the latest version. But when insertData3 comes around we'd have to change that into sql queries again, which will make the diff a bit uglier for that PR. So really just preventing future changes.

Reply:
Ah yes, I haven't thought about that!